### PR TITLE
ADX-762 VLS data for the ART table

### PR DESCRIPTION
**Pull request by jonathansberry**
Adding vls_tested and vls_suppression into the ART data template. 

I take it we are not going to enforce the `vls_tested > vls_supression` constraint?

### CODE REVIEW

File: table_schemas/art/4_art_estimates.json
Comment (jonathansberry) [inline table_schemas/art/4_art_estimates.json]: @jeffeaton Would be great if you can just confirm that I have got the right indicator definitions.

**Comment by jeffeaton** [comment]
Hi Jonathan,

Thanks — Eline / Mary are probably best to confirm. One thing to make sure to emphasise is routine VL tests only. We should also ensure the definitions are consistent with the Spectrum inputs.

I think we are still waiting for confirmation on the indicator definition. The definitions below are one option.  The other is “The number tested/suppressed for VL in the past 12 months” [This is the indicator definition used by PEPFAR].

If we go with the definitions below:

  *   vls_tested: “The number of people who received a routine viral load test during the reporting period.”
  *   For the vls_suppressed definition: “The number of people who were virally suppressed among those who received a routine viral load test during the period.”

One more question for all; should we use “vl_tested” and “vl_suppressed” for the indicator names?  “vls_tested” and “vls_suppressed” are a bit redundant; maybe it’s too much hassle to change it now though.

Thanks,
Jeff

**Comment by jonathansberry** [comment]
@jeffeaton 
- I've made the changes to the definitions you suggest and I will request Ian/Mary's approval before merging. If it's not dealt with by Wednesday we should try and finalise then.
- I agree that vl_tested vl_supressed makes more sense. It's easy to change column names on our side (one commit to this PR) if it's easy to change on yours.

File: table_schemas/art/4_art_estimates.json
Comment (jonathansberry) [inline table_schemas/art/4_art_estimates.json]: @wanyekii We're looking to pin down the exact indicator definitions for the new vls indicators in the ART template.  Can you chat with Mary and Eline and let me know if we're happy with the wording above? Jeff says (see comment above) that they should be well aligned with spectrum.

File: table_schemas/art/4_art_estimates.json
Comment (wanyekii) [reply table_schemas/art/4_art_estimates.json]: please enforce rule that the number tested should not be less than those who are suppressed

**Comment by jeffeaton** [comment]
From Ian:

Suggest we change “during the reporting period”to the last 12 months ie use “The number tested/suppressed for VL in the past 12 months”

This will match what is being entered into spectrum

Re vl vs vls, either can work from my end

**Comment by jeffeaton** [comment]
From Jonathan:

So…

The number of people who received a routine viral load test “during the 12 months prior to the reporting period end date”? 

And likewise for suppression?

**Comment by jeffeaton** [comment]
Hi all
Eline will be responding shortly.
she was just looking for a resource to add.

thanks
Mary


__________________________________
Mary Mahy, ScD, MHSc
Team Lead, Epidemiology
UNAIDS - Strategic Information Department
Geneva, Switzerland
+41 79 477 2103

**Comment by jeffeaton** [comment]
From Eline/Mary: 

Yes, yes we’d prefer to clarify as ‘the past 12 months’ instead of ‘the reporting period’.
Or even more precisely, as Jonathan just suggests: ‘the past 12 months prior to the reporting period end date’

File: table_schemas/art/4_art_estimates.json
Comment (jonathansberry) [reply table_schemas/art/4_art_estimates.json]: Done :+1:

**Comment by jonathansberry** [comment]
@jeffeaton Just waiting on a decision from Imperial as to whether we use `vls_` or `vl_` prefix.

**Review by wanyekii — DISMISSED** [review]
thanks looks good

**Comment by jeffeaton** [comment]
Two final questions from Jeff:

For the indicator names, would people be happy with vl_tested_12mos and vl_suppressed_12mos with the 12mos suffix to denote that the indicator is always for the 12 month period (even if reporting the indicator quarterly)

For the denominator for VL testing coverage, are we happy to use total currently on ART?  Or should that be lagged, e.g. number on ART 6 months earlier?   My interpretation of the GAM indicator sheet that Eline sent was it used currently on ART—which is much easier for us to plot. I think I have previously seen PEPFAR use 6 month lagged.

**Comment by jeffeaton** [comment]
I like the idea of being more specific with the indicator names, but will
defer to Eline and Mary for final confirmation before merging and deploying
the changes.

Best wishes,

Jonathan

**Comment by jeffeaton** [comment]
From Eline:
Agreed

**Comment by jeffeaton** [comment]
From Jeff:

Perfect, thanks. We will make the indicator change on our end now. It might take a few days to filter through the various bits of code.

Jonathan—I will send you an updated Antarctica test dataset later today.

**Comment by jeffeaton** [comment]
Jonathan – here is an updated Antarctica testing dataset with vl_tested_12mos and vl_suppressed_12mos.

**Comment by jeffeaton** [comment]
From Jonathan:

Great thanks, validation for ART and Shiny90 passes now in *staging* here:
https://dev.adr.fjelltopp.org/country-estimates-22/antarctica-country-estimates-2022-1-2

These changes will go to production with our next deployment.

**Pull request merged**
